subsys/caf/modules/sensor_manager.c

```diff
@@ -461,11 +461,15 @@ static bool handle_power_down_event(const struct event_header *eh)
 		 * and processing suspending if trigger was not configured.
 		 */
 		k_sched_lock();
-		if (sc->trigger) {
-			enter_sleep(sc, sd);
+		if (atomic_get(&sd->state) == SENSOR_STATE_ERROR) {
```

Comment:
Can we make this checking before the k_sched_lock? I would look nicer.

Reply:
Fix rewritten as discussed offline.